go/host/l1/blobresolver.go

```diff
@@ -37,21 +51,111 @@ func NewBlobResolver(beaconClient *ethadapter.L1BeaconClient, logger gethlog.Log
 }
 
 func (r *beaconBlobResolver) FetchBlobs(ctx context.Context, b *types.Header, hashes []gethcommon.Hash) ([]*kzg4844.Blob, error) {
-	var blobs []*kzg4844.Blob
+	// try fetching once to get the initial error and set appropriate retry strategy
+	blobs, initialErr := r.beaconClient.FetchBlobs(ctx, b, hashes)
+	if initialErr == nil {
+		return blobs, nil
+	}
 	err := retry.DoWithCount(func(retryNum int) error {
-		if retryNum > 5 {
-			r.logger.Info("Retrying to fetch blobs", "retryNum", retryNum)
-		}
 		var fetchErr error
 		blobs, fetchErr = r.beaconClient.FetchBlobs(ctx, b, hashes)
+
+		if fetchErr == nil {
+			return nil
+		}
+
+		retryStrategy := classifyError(fetchErr)
```

Comment:
Since you already picked the retry strategy based on the initial error (I think based on Tudor's comments) I'd probably remove this stuff here, it's quite confusing/misleading because the retry approach is already set.

So I mean maybe we can just log the error without classifying because the retry mechanism is already set.